lib/components/main_drawer.dart

```diff
@@ -1,48 +1,90 @@
-
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
-
+import 'package:geo_stories/models/user_dto.dart';
+import 'package:geo_stories/screens/edit_user_page.dart';
+import 'package:geo_stories/services/user_service.dart';
 import '../constants.dart';
 import 'Ui/user_circle_avatar.dart';
-class MainDrawer extends StatefulWidget {
 
+class MainDrawer extends StatefulWidget {
   _MainDrawerState createState() => _MainDrawerState();
 }
 
 class _MainDrawerState extends State<MainDrawer> {
+  User user;
+  bool userIsAuthenticated;
+  String userName;
+  String avatarURL;
+
+  @override
+  void initState() {
+    super.initState();
+    cargarDatos();
+  }
+
+  void cargarDatos() {
+    if (UserService.isAnonymousUser()) {
+      this.userIsAuthenticated = true;
+      this.user = UserService.getCurrentUser();
+      user.displayName != null ? this.userName = user.displayName : this.userName = "New User";
+      user.photoURL != null ? this.avatarURL = user.photoURL : this.avatarURL = kAvatarNotUser;
```

Comment:
Hacer una asignación de variables adentro de un ternario es turbio. Lo que podes hacer es:

`this.userName = user.DisplayName ?? "New User";`
`this.avatarURL = user.PhotoURL ?? kAvatarNotUser`